[BUG] cache_test failed w/ ParquetCachedBatchSerializer in spark 3.3.2-SNAPSHOT

**Describe the bug**
first saw at internal CI rapids_it-3.3.x-SNAPSHOT-dev-github build ID: 190

failed,
```

[2023-02-10T02:05:28.046Z] E                   Caused by: java.lang.IllegalArgumentException: For input string: "null"

[2023-02-10T02:05:28.046Z] E                   	at scala.collection.immutable.StringLike.parseBoolean(StringLike.scala:330)

[2023-02-10T02:05:28.046Z] E                   	at scala.collection.immutable.StringLike.toBoolean(StringLike.scala:289)

[2023-02-10T02:05:28.046Z] E                   	at scala.collection.immutable.StringLike.toBoolean$(StringLike.scala:289)

[2023-02-10T02:05:28.046Z] E                   	at scala.collection.immutable.StringOps.toBoolean(StringOps.scala:33)

[2023-02-10T02:05:28.046Z] E                   	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.<init>(ParquetSchemaConverter.scala:70)

[2023-02-10T02:05:28.046Z] E                   	at com.nvidia.spark.rapids.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.$anonfun$convertCachedBatchToInternalRowIter$1(ParquetCachedBatchSerializer.scala:744)

[2023-02-10T02:05:28.046Z] E                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)

[2023-02-10T02:05:28.046Z] E                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)

[2023-02-10T02:05:28.046Z] E                   	at com.nvidia.spark.rapids.ParquetCachedBatchSerializer.withResource(ParquetCachedBatchSerializer.scala:260)

[2023-02-10T02:05:28.046Z] E                   	at com.nvidia.spark.rapids.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.convertCachedBatchToInternalRowIter(ParquetCachedBatchSerializer.scala:729)

[2023-02-10T02:05:28.046Z] E                   	at com.nvidia.spark.rapids.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.hasNext(ParquetCachedBatchSerializer.scala:709)

[2023-02-10T02:05:28.046Z] E                   	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.hashAgg_doAggregateWithoutKey_0$(Unknown Source)

[2023-02-10T02:05:28.046Z] E                   	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)

[2023-02-10T02:05:28.046Z] E                   	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)

[2023-02-10T02:05:28.046Z] E                   	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)

[2023-02-10T02:05:28.047Z] E                   	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:140)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.scheduler.Task.run(Task.scala:136)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:548)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1504)

[2023-02-10T02:05:28.047Z] E                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:551)

[2023-02-10T02:05:28.047Z] E                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)

[2023-02-10T02:05:28.047Z] E                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
```

```
[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_expand_exec[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Float][IGNORE_ORDER, INCOMPAT]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_partial_load[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Array(String)][ALLOW_NON_GPU(CollectLimitExec)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_expand_exec[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Double][IGNORE_ORDER, INCOMPAT]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cached_join_filter[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Boolean][IGNORE_ORDER, ALLOW_NON_GPU(ANY)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_passing_gpuExpr_as_Expr[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'true'}][ALLOW_NON_GPU(CollectLimitExec)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_join[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Timestamp][IGNORE_ORDER]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_partial_load[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'true'}-Array(Decimal(20,2))][ALLOW_NON_GPU(CollectLimitExec)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_expand_exec[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Boolean][IGNORE_ORDER]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_partial_load[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Array(Boolean)][ALLOW_NON_GPU(CollectLimitExec)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cached_join_filter[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Date][IGNORE_ORDER, ALLOW_NON_GPU(ANY)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_expand_exec[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}-Date][IGNORE_ORDER]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_passing_gpuExpr_as_Expr[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}][ALLOW_NON_GPU(CollectLimitExec)]

[2023-02-10T02:05:28.047Z] FAILED ../../src/main/python/cache_test.py::test_cache_partial_load[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'true'}-String][ALLOW_NON_GPU(CollectLimitExec)]

...

=========== 476 failed, 104 passed, 48 warnings in 176.98s (0:02:56) ===========
```

**Steps/Code to reproduce bug**
```
  PYSP_TEST_spark_sql_cache_serializer=com.nvidia.spark.ParquetCachedBatchSerializer \
    ./run_pyspark_from_build.sh -k cache_test
```

--- a/pom.xml
+++ b/pom.xml
@@ -935,6 +935,7 @@
                                             <include name="311until320-noncdh/*"/>
                                             <include name="311until320-nondb/*"/>
                                             <include name="311until330-all/*"/>
+                                            <include name="311until332-all/*"/>
                                             <include name="311until330-nondb/*"/>
                                             <include name="311until340-all/*"/>
                                             <include name="311until340-non330db/*"/>
@@ -1035,7 +1036,9 @@
                                 <pathconvert property="spark332.sources" pathsep=",">
                                     <dirset dir="${project.basedir}/src/main" erroronmissingdir="false">
                                         <patternset refid="spark331+.pattern"/>
+                                        <exclude name="*until332*/*"/>
                                         <include name="332/*"/>
+                                        <include name="332+/*"/>
                                     </dirset>
                                 </pathconvert>
                                 <pathconvert property="spark340.sources" pathsep=",">

--- a/sql-plugin/src/main/311until332-all/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala
+++ b/sql-plugin/src/main/311until332-all/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022, NVIDIA CORPORATION.
+ * Copyright (c) 2023, NVIDIA CORPORATION.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/sql-plugin/src/main/311until332-all/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala
+++ b/sql-plugin/src/main/311until332-all/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala
@@ -1,0 +1,25 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+import org.apache.hadoop.conf.Configuration
+
+object ParquetLegacyNanoAsLongShims {
+  def setupLegacyParquetNanosAsLong(conf: Configuration): Unit = {
+    // Parquet Infer Timestamp NTZ is considered in Spark 3.3.2
+  }
+}

--- a/sql-plugin/src/main/311until332-all/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala
+++ b/sql-plugin/src/main/311until332-all/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala
@@ -19,7 +19,7 @@ package com.nvidia.spark.rapids.shims
 import org.apache.hadoop.conf.Configuration
 
 object ParquetLegacyNanoAsLongShims {
-  def setupLegacyParquetNanosAsLong(conf: Configuration): Unit = {
-    // Parquet Infer Timestamp NTZ is considered in Spark 3.3.2
+  def setupLegacyParquetNanosAsLongForPCBS(conf: Configuration): Unit = {
+    // LEGACY_PARQUET_NANOS_AS_LONG is only considered in Spark 3.3.2 and later
   }
 }

--- a/sql-plugin/src/main/332+/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala
+++ b/sql-plugin/src/main/332+/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2022, NVIDIA CORPORATION.
+ * Copyright (c) 2023, NVIDIA CORPORATION.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/sql-plugin/src/main/332+/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala
+++ b/sql-plugin/src/main/332+/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala
@@ -1,0 +1,27 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+import org.apache.hadoop.conf.Configuration
+
+import org.apache.spark.sql.internal.SQLConf
+
+object ParquetLegacyNanoAsLongShims {
+  def setupLegacyParquetNanosAsLong(conf: Configuration): Unit = {
+    conf.setBoolean(SQLConf.LEGACY_PARQUET_NANOS_AS_LONG.key, true)
+  }
+}

--- a/sql-plugin/src/main/332+/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala
+++ b/sql-plugin/src/main/332+/scala/com/nvidia/spark/rapids/shims/ParquetLegacyNanoAsLongShims.scala
@@ -21,7 +21,16 @@ import org.apache.hadoop.conf.Configuration
 import org.apache.spark.sql.internal.SQLConf
 
 object ParquetLegacyNanoAsLongShims {
-  def setupLegacyParquetNanosAsLong(conf: Configuration): Unit = {
+  /**
+   * This method should strictly be used by ParquetCachedBatchSerializer(PCBS) as it is hard coding
+   * the value of LEGACY_PARQUET_NANOS_AS_LONG.
+   *
+   * As far as PCBS is concerned it really doesn't matter what we set it to as long as
+   * ParquetSchemaConverter doesn't see a "null" value.
+   *
+   * @param conf Hadoop conf
+   */
+  def setupLegacyParquetNanosAsLongForPCBS(conf: Configuration): Unit = {
     conf.setBoolean(SQLConf.LEGACY_PARQUET_NANOS_AS_LONG.key, true)
   }
 }

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala
@@ -1202,7 +1202,7 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer wi
     ParquetFieldIdShims.setupParquetFieldIdWriteConfig(hadoopConf, sqlConf)
 
     // From 3.3.2, Spark schema converter needs this conf
-    ParquetLegacyNanoAsLongShims.setupLegacyParquetNanosAsLong(hadoopConf)
+    ParquetLegacyNanoAsLongShims.setupLegacyParquetNanosAsLongForPCBS(hadoopConf)
 
     hadoopConf
   }

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala
@@ -28,7 +28,7 @@ import ai.rapids.cudf.ParquetWriterOptions.StatisticsFrequency
 import com.nvidia.spark.GpuCachedBatchSerializer
 import com.nvidia.spark.rapids.GpuColumnVector.GpuColumnarBatchBuilder
 import com.nvidia.spark.rapids.RapidsPluginImplicits._
-import com.nvidia.spark.rapids.shims.{ParquetFieldIdShims, SparkShimImpl}
+import com.nvidia.spark.rapids.shims.{ParquetFieldIdShims, ParquetLegacyNanoAsLongShims, SparkShimImpl}
 import org.apache.commons.io.output.ByteArrayOutputStream
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.mapreduce.RecordWriter
@@ -477,7 +477,7 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer wi
           Table.readParquet(parquetOptions, parquetCB.buffer, 0, parquetCB.sizeInBytes)
         } catch {
           case e: Exception =>
-            throw new IOException("Error when processing file " + 
+            throw new IOException("Error when processing file " +
                 s"[range: 0-${parquetCB.sizeInBytes}]", e)
         }
         withResource(table) { table =>
@@ -557,7 +557,7 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer wi
       })
       cbRdd.mapPartitions(iter => CloseableColumnBatchIterator(iter))
     } else {
-      val origSelectedAttributesWithUnambiguousNames = 
+      val origSelectedAttributesWithUnambiguousNames =
         sanitizeColumnNames(newSelectedAttributes, selectedSchemaWithNames)
       val broadcastedConf = SparkSession.active.sparkContext.broadcast(conf.getAllConfs)
       input.mapPartitions {
@@ -1200,6 +1200,9 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer wi
 
     // From 3.3.0, Spark will check this filed ID config
     ParquetFieldIdShims.setupParquetFieldIdWriteConfig(hadoopConf, sqlConf)
+
+    // From 3.3.2, Schema converter
+    ParquetLegacyNanoAsLongShims.setupLegacyParquetNanosAsLong(hadoopConf)
 
     hadoopConf
   }

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020-2022, NVIDIA CORPORATION.
+ * Copyright (c) 2020-2023, NVIDIA CORPORATION.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -1201,7 +1201,7 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer wi
     // From 3.3.0, Spark will check this filed ID config
     ParquetFieldIdShims.setupParquetFieldIdWriteConfig(hadoopConf, sqlConf)
 
-    // From 3.3.2, Schema converter
+    // From 3.3.2, Spark schema converter needs this conf
     ParquetLegacyNanoAsLongShims.setupLegacyParquetNanosAsLong(hadoopConf)
 
     hadoopConf

